src/main/java/apoc/coll/Coll.java

```diff
@@ -431,6 +432,17 @@ public boolean containsAllSorted(@Name("coll") List<Object> coll, @Name("values"
         return true;
     }
 
+    @UserFunction
+    @Description("apoc.coll.isEqualCollection(coll, values) return true if two collections contain the same elements with the same cardinality in any order (using a HashMap)")
+    public boolean isEqualCollection(@Name("coll") List<Object> first, @Name("values") List<Object> second) {
+        if (first == null || first.isEmpty() || first.size() != second.size()) return false;
```

Comment:
We must manage the null check for the second otherwise we get a NullPointerException at line 442. 
I think that we can also get rid of `first.size() != second.size()` because we have the count comparison for each collection element.

Please also add a

Reply:
I did not do the null check on second because it was not done on the existing function I started from `containsAll` . Should I heck it there too ?
I added some tests for the various null/empty values.
I kept the starting size check as it allows exiting early without having to create the intermediate Maps, but if you prefer I'll remove it.

Reply:
In that case makes sense because:
1. there is a null check on `coll`
2. the `call.containsAll` method provides a null check internally.
Please add the null check.

Reply:
I could not find the check on call.containsAll:I added a test and `apoc.coll.containsAll([1,2,3],null) ` throws a NullPointerException